cmd/disk-cache.go

```diff
@@ -197,13 +194,18 @@ func (c cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string,
 		return c.GetObjectNInfoFn(ctx, bucket, object, rs, h, writeLock, opts)
 	}
 
-	backendDown := backendDownError(bkErr)
-	if bkErr != nil && !backendDown {
+	objInfo, err := c.GetObjectInfoFn(ctx, bucket, object, opts)
+	backendDown := backendDownError(err)
+	if err != nil && !backendDown {
```

Comment:
what is `backendDown` is `true`

Reply:
Good catch, then we can't rely on the contents of `objInfo` and therefore cannot check either `IsCachable(...)` or the other check below `filterFromCache(objInfo.UserDefined)`. The code structure here is a bit hard to read since an error is allowed as far as the backend is up.

Reply:
Did refactor `GetObjectNInfo` for readability, and also added the backend check to the depricated `GetObject`.